tests/unit/package_managers/bundler/test_parser.py

```diff
@@ -0,0 +1,278 @@
+from cachi2.core.package_managers.bundler.parser import (
+    GEMFILE,
+    GEMFILE_LOCK,
+    GemDependency,
+    GitDependency,
+    PathDependency,
+    parse_lockfile,
+)
+from cachi2.core.rooted_path import RootedPath
+
+
+@pytest.fixture
+def empty_gemfile(rooted_tmp_path: RootedPath) -> RootedPath:
+    gemfile_path = rooted_tmp_path.join_within_root(GEMFILE)
+    gemfile_path.path.touch()
+    return gemfile_path
+
+
+def test_parse_lockfile_without_lockfile(rooted_tmp_path: RootedPath) -> None:
+    msg = "Gemfile and Gemfile.lock must be present in the package directory"
+    with pytest.raises(PackageRejected, match=msg):
+        parse_lockfile(rooted_tmp_path)
+
+
+@mock.patch("cachi2.core.package_managers.bundler.parser.run_cmd")
+def test_parse_lockfile_os_error(
+    mock_run_cmd: mock.MagicMock,
+    empty_gemfile: RootedPath,
+    rooted_tmp_path: RootedPath,
+) -> None:
+    lockfile_path = rooted_tmp_path.join_within_root(GEMFILE_LOCK)
+    lockfile_path.path.touch()
+
+    mock_run_cmd.side_effect = subprocess.CalledProcessError(returncode=1, cmd="cmd")
+
+    with pytest.raises(PackageManagerError) as exc_info:
+        parse_lockfile(rooted_tmp_path)
+
+    assert f"Failed to parse {lockfile_path}" == str(exc_info.value)
+
+
+LOCKFILE_INVALID_URL = textwrap.dedent(
+    """
+    GIT
+      remote: github
+      revision: 26487618a68443e94d623bb585cb464b07d36702
+      specs:
+        json-schema (1.0.0)
+
+    PLATFORMS
+      ruby
+
+    DEPENDENCIES
+      json-schema!
+
+    BUNDLED WITH
+      2.5.10
+    """
+)
+
+LOCKFILE_INVALID_URL_SCHEME = textwrap.dedent(
+    """
+    GIT
+      remote: http://github.com/3scale/json-schema.git
+      revision: 26487618a68443e94d623bb585cb464b07d36702
+      specs:
+        json-schema (1.0.0)
+
+    PLATFORMS
+      ruby
+
+    DEPENDENCIES
+      json-schema!
+
+    BUNDLED WITH
+      2.5.10
+    """
+)
+
+LOCKFILE_INVALID_REVISION = textwrap.dedent(
+    """
+    GIT
+      remote: https://github.com/3scale/json-schema.git
+      revision: abcd1234
+      specs:
+        json-schema (1.0.0)
+
+    PLATFORMS
+      ruby
+
+    DEPENDENCIES
+      json-schema!
+
+    BUNDLED WITH
+      2.5.10
+    """
+)
+
+LOCKFILE_INVALID_PATH = textwrap.dedent(
+    """
+    PATH
+      remote: /root/pathgem
+      specs:
+        pathgem (1.0.0)
+
+    PLATFORMS
+      ruby
+
+    DEPENDENCIES
+      pathgem!
+
+    BUNDLED WITH
+      2.5.10
+    """
+)
+
+
+@pytest.mark.parametrize(
+    "lockfile_content, expected_error",
+    [
+        (
+            LOCKFILE_INVALID_URL,
+            "Input should be a valid URL",
+        ),
+        (
+            LOCKFILE_INVALID_URL_SCHEME,
+            "URL scheme should be 'https'",
+        ),
+        (
+            LOCKFILE_INVALID_REVISION,
+            "String should match pattern '^[a-fA-F0-9]{40}$'",
+        ),
+        (
+            LOCKFILE_INVALID_PATH,
+            "PATH dependencies should be within the package root",
+        ),
```

Comment:
Is this black's work? We should be able to fit individual params on a single line, right?

Reply:
That was caused by a comma at the end of the string.